test/metabase/public_settings_test.clj

```diff
@@ -350,3 +350,21 @@
            Exception
            #"This field must be a relative URL."
            (public-settings/landing-page! "file:///path/to/resource")))))
+
+(deftest show-metabase-links-test
+  (mt/with-premium-features #{:whitelabel}
+    (testing "When whitelabeling is enabled, show-metabase-links setting can be set to boolean"
+      (public-settings/show-metabase-links! true)
+      (is (= true (public-settings/show-metabase-links)))
+
+      (public-settings/show-metabase-links! false)
+      (is (= false (public-settings/show-metabase-links)))))
+
+  (mt/with-premium-features #{}
+    (testing "When whitelabeling is not enabled, show-metabase-links setting cannot be set, and always returns true"
+      (is (thrown-with-msg?
+           clojure.lang.ExceptionInfo
+           #"Setting show-metabase-links is not enabled because feature :whitelabel is not available"
+           (public-settings/show-metabase-links! true)))
+
+      (is (= true (public-settings/show-metabase-links))))))
```

Comment:
nit — you can use `discard-setting-changes` to ensure that the state gets reset at the end of the test
```suggestion
(deftest help-link-setting-test
  (mt/discard-setting-changes [help-link]
    (mt/with-premium-features #{:whitelabel}
      (testing "When whitelabeling is enabled, help-link setting can be set to any valid value"
        (public-settings/help-link! :metabase)
        (is (= :metabase (public-settings/help-link)))

        (public-settings/help-link! :hidden)
        (is (= :hidden (public-settings/help-link)))

        (public-settings/help-link! :custom)
        (is (= :custom (public-settings/help-link))))

      (testing "help-link cannot be set to an invalid value"
        (is (thrown-with-msg?
             Exception #"Invalid help link option"
             (public-settings/help-link! :invalid)))))

    (mt/with-premium-features #{}
      (testing "When whitelabeling is not enabled, help-link setting cannot be set, and always returns :metabase"
        (is (thrown-with-msg?
             clojure.lang.ExceptionInfo
             #"Setting help-link is not enabled because feature :whitelabel is not available"
             (public-settings/help-link! :hidden)))

        (is (= :metabase (public-settings/help-link)))))))
```

Reply:
This is new to me. Thanks for the feedback! Fixed in cd53e85a47714f7e57f9ea1cefbf8aee8a185005